### ISSUE TITLE
Undefined array key "type"

### ISSUE THREAD

**Issue by anchovy** [verbb/vizy#168]
### Describe the bug

I've got nestled vizy editors. When I add 2 or more blocks to a nestled editor I get the exception: Undefined array key "type" in /home/forge/dev.nzda.org.nz/vendor/verbb/vizy/src/models/NodeCollection.php:213

Stack trace:
```
#0 /craftsite/vendor/craftcms/cms/src/web/ErrorHandler.php(85): yii\base\ErrorHandler->handleError()
#1 /craftsite/vendor/verbb/vizy/src/models/NodeCollection.php(213): craft\web\ErrorHandler->handleError()
#2 /craftsite/vendor/verbb/vizy/src/models/NodeCollection.php(51): verbb\vizy\models\NodeCollection->_populateNodes()
#3 /craftsite/vendor/verbb/vizy/src/fields/VizyField.php(277): verbb\vizy\models\NodeCollection->__construct()
#4 /craftsite/vendor/craftcms/cms/src/base/Element.php(4296): verbb\vizy\fields\VizyField->normalizeValue()
#5 /craftsite/vendor/craftcms/cms/src/base/Element.php(3355): craft\base\Element->normalizeFieldValue()
#6 /craftsite/vendor/craftcms/cms/src/base/Element.php(1759): craft\base\Element->getFieldValue()
#7 /craftsite/vendor/yiisoft/yii2/validators/InlineValidator.php(82): craft\base\Element->__get()
#8 /craftsite/vendor/yiisoft/yii2/validators/Validator.php(261): yii\validators\InlineValidator->validateAttribute()
#9 /craftsite/vendor/yiisoft/yii2/base/Model.php(367): yii\validators\Validator->validateAttributes()
#10 /craftsite/vendor/verbb/vizy/src/fields/VizyField.php(554): yii\base\Model->validate()
#11 /craftsite/vendor/craftcms/cms/src/base/Element.php(2123): verbb\vizy\fields\VizyField->validateBlocks()
#12 /craftsite/vendor/yiisoft/yii2/validators/InlineValidator.php(84): craft\base\Element->validateCustomFieldAttribute()
#13 /craftsite/vendor/yiisoft/yii2/validators/Validator.php(261): yii\validators\InlineValidator->validateAttribute()
#14 /craftsite/vendor/yiisoft/yii2/base/Model.php(367): yii\validators\Validator->validateAttributes()
#15 /craftsite/vendor/verbb/super-table/src/fields/SuperTableField.php(833): yii\base\Model->validate()
#16 /craftsite/vendor/craftcms/cms/src/base/Element.php(2123): verbb\supertable\fields\SuperTableField->validateBlocks()
#17 /craftsite/vendor/yiisoft/yii2/validators/InlineValidator.php(84): craft\base\Element->validateCustomFieldAttribute()
#18 /craftsite/vendor/yiisoft/yii2/validators/Validator.php(261): yii\validators\InlineValidator->validateAttribute()
#19 /craftsite/vendor/yiisoft/yii2/base/Model.php(367): yii\validators\Validator->validateAttributes()
#20 /craftsite/vendor/craftcms/cms/src/services/Elements.php(2552): yii\base\Model->validate()
#21 /craftsite/vendor/craftcms/cms/src/services/Elements.php(785): craft\services\Elements->_saveElementInternal()
#22 /craftsite/vendor/craftcms/cms/src/controllers/EntryRevisionsController.php(313): craft\services\Elements->saveElement()
#23 [internal function]: craft\controllers\EntryRevisionsController->actionSaveDraft()
#24 /craftsite/vendor/yiisoft/yii2/base/InlineAction.php(57): call_user_func_array()
#25 /craftsite/vendor/yiisoft/yii2/base/Controller.php(178): yii\base\InlineAction->runWithParams()
#26 /craftsite/vendor/yiisoft/yii2/base/Module.php(552): yii\base\Controller->runAction()
#27 /craftsite/vendor/craftcms/cms/src/web/Application.php(293): yii\base\Module->runAction()
#28 /craftsite/vendor/craftcms/cms/src/web/Application.php(602): craft\web\Application->runAction()
#29 /craftsite/vendor/craftcms/cms/src/web/Application.php(272): craft\web\Application->_processActionRequest()
#30 /craftsite/vendor/yiisoft/yii2/base/Application.php(384): craft\web\Application->handleRequest()
#31 /craftsite/web/index.php(26): yii\base\Application->run()
#32 {main}
```

### Steps to reproduce

1. 
2. 

### Craft CMS version

Craft Pro 3.7.50

### Plugin version

1.0.17

### Multi-site?

No

### Additional context

With a single block in the nestled editor it's fine: 
![Screenshot 2022-08-09 at 1 21 24 PM](https://user-images.githubusercontent.com/599005/183542189-d970473d-ff10-4375-9a39-89115812a041.png)

2x blocks it throws the error
![Screenshot 2022-08-09 at 1 20 46 PM](https://user-images.githubusercontent.com/599005/183542180-587f9240-1c12-4869-9f9c-c1a427556cf5.png)

**Comment by engram-design** [comment]
I can't seem to replicate this with all manner of nested field setups. But this error tells me that the node is missing its `type` property, which should be impossible. I'll keep looking into it.

**Comment by anchovy** [comment]
Hi @engram-design - have you had any other ideas on this? Any other debugging I can do to help?

**Comment by engram-design** [comment]
No other bright ideas I'm afraid, other than _something_ must've gotten messed up with your fields on your install. I can certainly fix the hard error being thrown, but it's probably just bandaid-ing the real issue where a node has lost the definition of what type it is.

If you want to debug, you can insert the following code here https://github.com/verbb/vizy/blob/bebd780553e00893e10e2140fb96f08001184127/src/models/NodeCollection.php#L212

```php
if (!isset($node['type'])) {
    Craft::dd($node);
}
```

Which will at least print out what `$node` contains, if not its type.

This will produce a hard error that you might need to use your network tools to inspect the request with.

**Comment by anchovy** [comment]
Hi @engram-design thanks for that.

Issue is still occurring, here's the hard errror I get inspecting the request:

```
[
    'vizy-block-Aa8FQzo63A' => [
        'fields' => [
            'visy_image' => [
                'sortOrder' => [
                    0 => 'new1'
                ]
                'blocks' => [
                    'new1' => [
                        'type' => 'image'
                        'enabled' => '1'
                        'fields' => [
                            'image' => [
                                0 => '30987'
                            ]
                            'image_link' => [
                                'type' => 'entry'
                                'url' => ''
                                'custom' => ''
                                'email' => ''
                                'tel' => ''
                                'entry' => [
                                    'value' => ''
                                ]
                                'target' => '_blank'
                                'title' => ''
                                'isCpFormData' => 'true'
                            ]
                            'topMargin' => 'base64:bXQtMA=='
                            'bottomMargin' => 'base64:bWItMA=='
                            'maximumWidth' => ''
                            'width' => [
                                'locale' => 'en-GB'
                                'value' => '140'
                            ]
                            'altTag' => ''
                            'alignment' => 'base64:Y2VudHJl'
                        ]
                    ]
                ]
            ]
        ]
    ]
]
```

its occurring with very simple blocks as well i.e.
```
[
    'vizy-block-iskjPF2mD3' => [
        'fields' => [
            'spacer' => [
                'locale' => 'en-GB'
                'value' => '50'
            ]
        ]
    ]
]
```

**Comment by engram-design** [comment]
Thanks for that. Can you confirm your full field setup and types, so I can try and replicate your exact fields?

**Comment by anchovy** [comment]
Hi @engram-design - I've got a stripped version of our site that I could share with you which exhibits this error? Ok to add you to the repo?

**Comment by engram-design** [comment]
@anchovy Yes please!

**Comment by anchovy** [comment]
@engram-design - just sent you an invite to the private repo. Will send you _your_ login details via an issue on the repo when you accept.

**Comment by engram-design** [comment]
Thanks! I should be able to get started without login details (I'll just create them)

**Comment by engram-design** [comment]
Thanks for that info @anchovy looks like a bit of a legacy issue with the older version of Vizy (for Craft 3). Should be fixed for the next release. To get the fix early, change your `verbb/vizy` requirement in `composer.json` to:

```
"require": {
  "verbb/vizy": "dev-craft-3 as 1.0.18",
  "...": "..."
}
```

Then run `composer update`.

**Comment by anchovy** [comment]
Hi @engram-design. With this release installed it's not rendering any of the blocks at all? 
Updated the issue ticket on the private repo with a recording of what it's doing now.

**Comment by engram-design** [comment]
Fixed in [1.0.20](https://github.com/verbb/vizy/releases/tag/1.0.20)

**Comment by denisyilmaz** [comment]
I'm experiencing this error now again in Craft 4.2.8 with Vizy 2.0.8
